### PR TITLE
Adding Contributing Guide and Fixed Readme

### PR DESCRIPTION
**Pull request by Haimantika**
It closes issue - https://github.com/khattakdev/minimalcommit/issues/6

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+- Comment on the issue expressing your interest.
+- Mention how you would like to approach the issue
+- Wait for a response from the maintainers of the project. 
+
+## Next Steps
+Once the issue is assigned to you, it's time to start working on it. Make sure to adhere to our coding and documentation standards.Keep your assigned issue updated with your progress and any challenges you encounter. If you're unable to continue working on the issue for any reason, please let us know as soon as possible.
+
+## Request to Work on Stalled Issues
+If an issue has been assigned to someone else and there have been no updates or activity for more than two weeks, you may request to work on it. This helps keep the project moving and ensures that issues are not left unattended for extended periods.
```
Comment (khattakdev) [inline CONTRIBUTING.md]: Since it is a small project, change `two weeks` to `a week`

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+If you find an existing issue that you'd like to work on and it has not been assigned to anyone, you can request ownership of the issue. 
+
+Here's how to do it:
+
+- Comment on the issue expressing your interest.
+- Mention how you would like to approach the issue
+- Wait for a response from the maintainers of the project. 
+
+## Next Steps
+Once the issue is assigned to you, it's time to start working on it. Make sure to adhere to our coding and documentation standards.Keep your assigned issue updated with your progress and any challenges you encounter. If you're unable to continue working on the issue for any reason, please let us know as soon as possible.
```
Comment (khattakdev) [inline CONTRIBUTING.md]: Add spacing to `standards.Keep` -> `standards. Keep`

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,35 @@
+Thank you for considering contributing to Minimal Commit! Your contributions are valuable in helping us improve and grow. To ensure a smooth and collaborative development process, please follow these guidelines when working on issues:
+
```
Comment (khattakdev) [inline CONTRIBUTING.md:51]: Add a **Setup** section highlighting steps to set up the repo locally. Also, add **Prerequisites**, highlighting what's needed to work on the project (`Nodejs)`

Comment (Haimantika) [reply CONTRIBUTING.md:51]: Fixed it in the recent commit.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,77 @@
+Thank you for considering contributing to Minimal Commit! Your contributions are valuable in helping us improve and grow. 
+
+## Prerequisites
+Before you start contributing to the project, here's the prerequisities that you need to have:
+
+1. **Git:** You'll need Git to clone the repository and manage version control. If Git is not installed, download and install it from [Git Downloads](https://git-scm.com/downloads).
+
+2. **Node.js:** The project is built using Node.js. You can download and install Node.js from [Node.js Downloads](https://nodejs.org/en/download/). Make sure to install a LTS (Long Term Support) version.
+
+3. **npm (Node Package Manager):** npm is usually bundled with Node.js. To check if you have npm installed, run the following command in your terminal: `npm v`
```
Comment (khattakdev) [inline CONTRIBUTING.md]:
```suggestion
3. **npm (Node Package Manager):** npm is usually bundled with Node.js. To check if you have npm installed, run the following command in your terminal: `npm -v` 
```

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,77 @@
+Thank you for considering contributing to Minimal Commit! Your contributions are valuable in helping us improve and grow. 
+
+## Prerequisites
+Before you start contributing to the project, here's the prerequisities that you need to have:
+
+1. **Git:** You'll need Git to clone the repository and manage version control. If Git is not installed, download and install it from [Git Downloads](https://git-scm.com/downloads).
```
Comment (khattakdev) [inline CONTRIBUTING.md]:
```suggestion
1. **Git:** You'll need git to clone the repository and manage version control. If git is not installed, download and install it from [Git Downloads](https://git-scm.com/downloads).
```

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,77 @@
...
+Open your terminal and navigate to the directory where you want to store the project. Then, run the following command to clone the repository: `git clone https://github.com/yourusername/minimalcommit.git`
+
+
+3. **Change Directory:**
+Navigate into the project directory: `cd minimalcommit`
+
+4. **Install Dependencies:**
+Use npm to install the project's dependencies: `npm install`
+
+5. **Configure Environment Variables:**
```
Comment (khattakdev) [inline CONTRIBUTING.md]: Remove this part - we don't have any `.env` configurations for now.

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+The project may require environment variables to be set for configuration. Check if there is an `.env` file or any specific instructions in the project's documentation. If required, create an `.env` file and add the necessary variables.
+
+6. **Start the Development Server:**
+To start the development server, run the following command: `npm start`
```
Comment (khattakdev) [inline CONTRIBUTING.md]: This is a Node.js project. There isn't any `npm start` command.

Two ways to test the project
- `node src/cli.js`
- Linked the npm project locally using `npm link` command ([Learn more](https://docs.npmjs.com/cli/v9/commands/npm-link)

**Review by khattakdev — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md (hunk 7 above)
Comment (Haimantika) [reply CONTRIBUTING.md]: Worked on it. Not sure if it is the best way to write.

File: CONTRIBUTING.md [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+There are two ways to test the project:
+
+- **Run the CLI :** You can test the project by running the CLI using the following command:
+  ```
+  node src/cli.js
+  ```
+
+- **Link the Project Locally:** To test the project locally as a global command, you can use the `npm link` command. First, navigate to the project directory and run the following command:
+  ```
+  npm link
```
Comment (khattakdev) [inline CONTRIBUTING.md]:
```suggestion
  npm link minimalcommit
```

**Review by khattakdev — APPROVED** [review]

**Pull request merged**